#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "Floor"
      material2 "Smooth"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.752209744582965 -0.6236434623198179 -0.21271890386183465 0.8018949709431734
  position -0.25341791922612217 0.46942855831555136 0.5813788631610797
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  contactMaterial "Floor"
  floorTileSize 0.2 0.2
}
Robot {
  translation 0 0.12 0
  children [
    GPS {
    }
    DEF Body Group {
      children [
        Transform {
          translation -0.06 -0.03 0
          children [
            DEF Ball Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Sphere {
                radius 0.02
                subdivision 2
              }
            }
          ]
        }
        DEF Box Shape {
          appearance DEF Body_App PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.2 0.06 0.1
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2.5253684986159427e-09
        axis 0 0 -1
        anchor 0.05 -0.02 0.06
      }
      device [
        DEF Wheel1 RotationalMotor {
          name "RightWheelMotor"
          maxVelocity 100
        }
      ]
      endPoint Solid {
        translation 0.050236433751432594 -0.02013847202816315 0.06029079645146594
        rotation 0.9781056539555855 0.15526188894798448 0.13857516206167736 1.5776131286985284
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.035
            }
          }
        ]
        contactMaterial "Wheel"
        boundingObject USE Wheel
        physics DEF Physics_Wheel Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2.525400603234157e-09
        axis 0 0 -1
        anchor 0.05 -0.02 -0.06
      }
      device [
        DEF Wheel2 RotationalMotor {
          name "LeftWheelMotor"
          maxVelocity 100
        }
      ]
      endPoint Solid {
        translation 0.05010010066489293 -0.019743767119615983 -0.06028681898865934
        rotation 0.9681992927225499 0.16828191005618295 0.1851251693298267 1.5845924886457197
        children [
          USE Wheel
        ]
        name "solid(1)"
        contactMaterial "Wheel"
        boundingObject USE Wheel
        physics USE Physics_Wheel
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -1.8326478135820424e-07
        axis 0 0 1
        anchor 0.09 0.025 0
      }
      device [
        RotationalMotor {
          name "ClawMotor"
        }
      ]
      endPoint Solid {
        translation 0.22592290966235515 -5.020247630986227e-05 1.0510479863307108e-05
        rotation -0.9381703480922411 -0.11796676375733871 0.3254538993607293 1.4421244954725775e-05
        children [
          LightSensor {
            translation 0.007 -0.02 0.002
            children [
              Shape {
                appearance PBRAppearance {
                }
                geometry Box {
                  size 0.0018 0.0041 0.0034
                }
              }
            ]
            name "light sensor left"
          }
          LightSensor {
            translation 0.007 -0.02 -0.002
            children [
              Shape {
                appearance PBRAppearance {
                }
                geometry Box {
                  size 0.0018 0.0041 0.0034
                }
              }
            ]
            name "light sensor right"
          }
          DistanceSensor {
            translation 0.001 -0.01 0
            rotation 0 1 0 4.07987e-06
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 1 1
                  metalness 0
                }
                geometry Box {
                  size 0.014 0.014 0.03
                }
              }
            ]
            name "DistanceSensorFront"
          }
          DEF Holder Group {
            children [
              Shape {
                appearance DEF Holder_App PBRAppearance {
                  baseColor 1 0 1
                  metalness 0
                }
                geometry Box {
                  size 0.015 0.06 0.1
                }
              }
              Transform {
                translation -0.045 0 -0.0575
                children [
                  Shape {
                    appearance USE Holder_App
                    geometry Box {
                      size 0.105 0.06 0.015
                    }
                  }
                ]
              }
              Transform {
                translation -0.12 0.0225 -0.0575
                children [
                  Shape {
                    appearance USE Holder_App
                    geometry Box {
                      size 0.045 0.015 0.015
                    }
                  }
                ]
              }
              Transform {
                translation -0.12 0.0225 0.0575
                children [
                  Shape {
                    appearance USE Holder_App
                    geometry Box {
                      size 0.045 0.015 0.015
                    }
                  }
                ]
              }
              Transform {
                translation -0.045 0 0.0575
                children [
                  Shape {
                    appearance USE Holder_App
                    geometry Box {
                      size 0.105 0.06 0.015
                    }
                  }
                ]
              }
            ]
          }
        ]
        name "solid(2)"
        boundingObject USE Holder
        physics Physics {
        }
      }
    }
  ]
  name "robot(1)"
  contactMaterial "Smooth"
  boundingObject USE Body
  physics DEF Physics_Body Physics {
    mass 0.8
  }
  controller "JASMINE_controller"
}
